### DISCUSSION TITLE
Feedback regarding the City Community "New Event" page

### DISCUSSION THREAD

**Discussion by rahulporuri** [fossunited/fossunited#324] [Q&A] [answered]
![Screenshot 2024-05-15 073918](https://github.com/fossunited/fossunited/assets/1926457/b84009d2-846e-479a-9cde-64901bad9cf3)

* Looking at the page, it wasn't obvious what was a necessary field and what is not a necessary field
* I tried creating the event without "Event Start Date" and "Event End Date" and I got a "Failed to create event" error on the bottom-right corner of the screen. It wasn't obvious how to fix the error but when I looked into the console, I noticed the following errors
<details><summary>See full traceback below</summary>
```
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 110, in application
    response = frappe.api.handle(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/__init__.py", line 49, in handle
    data = endpoint(**arguments)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/api/v1.py", line 36, in handle_rpc_call
    return frappe.handler.handle()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 49, in handle
    data = execute_cmd(cmd)
           ^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/handler.py", line 85, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1768, in call
    return fn(*args, **newargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/client.py", line 201, in insert
    return insert_doc(doc).as_dict()
           ^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/client.py", line 485, in insert_doc
    return frappe.get_doc(doc).insert()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 292, in insert
    self._validate()
  File "apps/frappe/frappe/model/document.py", line 574, in _validate
    self._validate_mandatory()
  File "apps/frappe/frappe/model/document.py", line 911, in _validate_mandatory
    raise frappe.MandatoryError(
frappe.exceptions.MandatoryError: [FOSS Chapter Event, immbjou1p7]: event_start_date, event_end_date
```
</details>

![Screenshot 2024-05-15 081926](https://github.com/fossunited/fossunited/assets/1926457/f10e3c8e-4c09-4d8e-ba1b-0bddfec08291)

* I was expecting a sensible default for the "Event End Date" after the "Event Start Date" value was set to a certain datetime.
* It's not clear what the difference between the "Being Reviewed" and "Accepted" Event Status is
![Screenshot 2024-05-15 082330](https://github.com/fossunited/fossunited/assets/1926457/28b6059f-31fa-4b98-ac27-0ac5cdd2acd7)
* I defined the event permalink as `hyd/2024/jun` because it wasn't clear what characters were allowed and what are not allowed. I was able to successfully create the event but the link is broken https://fossunited.org/events/hyd/2024/jun when i attempt to open the event https://fossunited.org/hyderabad
* I went back and updated the event permalink to `hyd-2024-jun` but when I attempt to open the event page, it still points to the broken page - https://fossunited.org/hyderabad
* On the "Manage Event" page, under "RSVP" > "Web Form" - "Route of the RSPV form". The spelling needs to be corrected to "RSVP", not "RSPV"
* Clicking on the "Add Volunteer" button on the "Volunteers" page creates a modal dialog that uses the text "Add New Member". It would have been better to use "Add New Volunteer"
* The "search for a user" dropdown provides an option to "Select All". I'm not sure why any city chapter would like use the "Select All" button

**Comment by ansharora28** [comment]
> link is broken

We're supposed to publish the event as well. I've done it now.
https://fossunited.org/events/hyd-2024-jun

Maybe this should be enabled by default? I don't think there is any use-case for creating an event and not publishing it :thinking: I was pretty confused with this too when creating the delhi meetup last week
It should be checked by default with the option to uncheck it if we want to unpublish
![image](https://github.com/fossunited/fossunited/assets/89684245/a46f2ccb-9046-4a42-b911-c0c8ad593b2b)

**Reply by harshtandiya** [reply]
Makes sense to publish it by default. I will create an Issue and work on it

**Reply by harshtandiya** [reply]
https://github.com/fossunited/fossunited/issues/325

**Reply by rahulporuri** [reply]
> We're supposed to publish the event as well. I've done it now.

I'm sorry but who is "We"? I don't see the "Meta Info" > "Is Published?" toggle on the Create Event page or the Event page.

**Reply by ansharora28** [reply]
> I don't see the "Meta Info" > "Is Published?" toggle on the Create Event page or the Event page.

Sorry, I just realised the meta info section only exists on frappe desk. @harshtandiya this pretty much makes publishing events through the dashboard impossible - can we add this to the dashboard too please?
![image](https://github.com/fossunited/fossunited/assets/89684245/89895f9c-ea6a-42f6-8f24-99738845815a)

**Reply by rahulporuri** [reply]
> can we add this to the dashboard too please?

So, if I understand correctly, is this the recommended process

* A city chapter considers hosting a monthly meetup
* A city chapter volunteer creates a "Being Reviewed" event as a placeholder. The event isn't visible publicly at this point
* The city chapter lead/team then confirms the date, location, and other information etc and "Approves" the meetup, which causes it to become publicly visible

Was that the expected workflow?

**Reply by ansharora28** [reply]
That sounds right. In some cases, you'd also want to create an event in advance and make it public once everything is finalised ( the venue gives a final go ahead etc.)

**Reply by harshtandiya** [reply]
Making the process more streamlined now. The event will be published when Status is Live or Concluded. When it is draft or cancelled, it will not be published. I would want to remove the requirement of someone from our team constantly reviewing events.

**Reply by RiyaMathew-11** [reply, marked as answer]
@harshtandiya 

Let's also mark the essential fields in asterix and check on other issues in mind too. 

Referencing this discussion - i've made an issue with a list - #334

These could be made individual issues as per need.

**Comment by RiyaMathew-11** [comment]
Marking this as answered since with the relevant issue list created.
